drivers/sensor/Kconfig

```diff
@@ -24,6 +24,8 @@ config SENSOR_ASYNC_API
 	bool "Async Sensor API"
 	select RTIO
 	select RTIO_SYS_MEM_BLOCKS
+	select RTIO_CONSUME_SEM
+	select RTIO_WORKQ
```

Comment:
These options should likely be done at a driver level, e.g. sensor/bme280/Kconfig should select it

Reply:
That's true - In this case, though:
- `RTIO_CONSUME_SEM` is needed even with no `RTIO_WORKQ`  so that `sensor_processing_with_callback` does not block using k_yield(), which could be a problem depending on thread priorities.
- `RTIO_WORKQ` is needed for `sensor_submit_fallback`.

As an action item, I've added `RTIO_WORKQ` to each individual sensor driver so that when the sensor shim goes, it'll be up to each driver to scrub it on its own time.

Reply:
Fair points! It’s fine then